### PR TITLE
Adds support for optionally randomizing Nullable<T> to sometimes return null

### PR DESCRIPTION
**Pull request by tystol**
Another small addition I needed for my test purposes.

Again, I left existing behavior unchanged and made it opt-in.

What I don't like though is the static property that controls the behavior (thread-safety. Technically it's still thread-safe, just that nullable behavior may change mid execution). A nice future addition would be making the class non-static, allowing per-instance settings for controlling behavior.

### CODE REVIEW

**Comment by Sobieck** [comment]
I have a couple ideas about making this thread safe and not having to add a flag to indicate at a class level that you want the nullable values to sometimes return null. 

1. Create a new call on RandomValue. Maybe `RandomValue.ObjectWithNullableValues<T>();`
2. Create an overload that takes a flag to replace `RandomValue.IncludeNullAsPossibleValueForNullables`. Something like RandomValue.Object<T>(includeNullableValues).

Basically I would like to avoid managing state at the class level and pass the flag as an argument or a default on another call. 

What are your thoughts?

**Comment by tystol** [comment]
I thought about that, but it quickly spreads through many of the methods as it needs to be obeyed recursively down the object graph (similar to my previous PR's `recursiveDepth` parameter).

If this library keeps evolving, ultimately passing each config setting around via method parameters becomes unwieldy. Hence the need for instance level methods and settings (or keep it static, and pass a config object around recursively). But I didn't want to make that refactor part of this PR.

**Comment by RasicN** [comment]
Is this something I still need to merge?  I apologize for just now looking at this!

**Comment by Sobieck** [comment]
I solved this issue with 369a8599baf5bfd71e1263dc9e7fddfbf2a57283. That feature is now part of the settings object which I don't think we've documented...
